### PR TITLE
tests: Add a test for layouts for 2D views of 3D image slices

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Add a test for image layout transitions of
VK_IMAGE_CREATE_2D_ARRAY_COMPATIBLE_BIT images.

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
This PR will fix  #2851 by providing a test case that shows the regression, even though it was subsequently fixed by changes between the 1.2.176.1 SDK and current tip of tree.

File: tests/vkpositivelayertests.cpp [hunk 1]
```diff
@@ -12904,6 +12904,238 @@ TEST_F(VkPositiveLayerTest, DevsimLoaderCrash) {
...
+        kExternal   // Image layout mismatch is with the current state of the image, found at QueueSubmit
+    };
+    std::array<TestType, 2> test_list = {{kInternal, kExternal}};
+
+    auto do_test = [&](VkImageObj *image, vk_testing::ImageView *view, VkImageObj *o_image, vk_testing::ImageView *o_view, VkImageAspectFlags aspect_mask,
+                       VkImageLayout descriptor_layout) {
+        // Set up the descriptor
+        img_info.imageView = o_view->handle();
+        img_info.imageLayout = descriptor_layout;
+        vk::UpdateDescriptorSets(m_device->device(), 1, &descriptor_write, 0, NULL);
```
Comment (Tony-LunarG) [inline tests/vkpositivelayertests.cpp:13067]: I think this gets 00343 - imageView must not be 2D or 2D array image view created from a 3D image

**Review by Tony-LunarG — CHANGES_REQUESTED** [review]
I'm seeing some unexpected errors in the setup code

**Comment by jeremyg-lunarg** [comment]
> I'm seeing some unexpected errors in the setup code

Yep once I moved the ExpectSuccess I got them too. Should be fixed now, Thanks!

**Pull request merged**